cli/config/configfile/file.go

```diff
@@ -36,6 +36,10 @@ type ConfigFile struct {
 	Filename             string                      `json:"-"` // Note: for internal use only
 	ServiceInspectFormat string                      `json:"serviceInspectFormat,omitempty"`
 	ServicesFormat       string                      `json:"servicesFormat,omitempty"`
+	HTTPProxy            string                      `json:"httpProxy,omitempty"`
```

Comment:
I've changed this (patch set coming soon) to:

```
{ 
  "proxies": 
    {
      "default":
        {
          "httpProxy" : "http://proxy.example.com"
        }
     }
}
```

These will now be used for both `docker build` and `docker run`.

The top-level key can be easily changed to reflect this.
@alexellis suggested `containerProxies` but I'm undecided as yet